### PR TITLE
Cairo0 hint uint256 sqrt

### PR DESCRIPTION
**Pull request by MaksymMalicki**

### CODE REVIEW

File: pkg/hintrunner/zero/hintcode.go [hunk 1]
```diff
@@ -38,4 +38,5 @@ const (
 	uint256AddCode    string = "sum_low = ids.a.low + ids.b.low\nids.carry_low = 1 if sum_low >= ids.SHIFT else 0\nsum_high = ids.a.high + ids.b.high + ids.carry_low\nids.carry_high = 1 if sum_high >= ids.SHIFT else 0"
 	uint256AddLowCode string = "sum_low = ids.a.low + ids.b.low\nids.carry_low = 1 if sum_low >= ids.SHIFT else 0"
 	split64Code       string = "ids.low = ids.a & ((1<<64) - 1)\nids.high = ids.a >> 64"
+	uint256Sqrt       string = "from starkware.python.math_utils import isqrt\nn = (ids.n.high << 128) + ids.n.low\nroot = isqrt(n)\nassert 0 <= root < 2 ** 128\nids.root.low = root\nids.root.high = 0"
```
Comment (quasilyte) [inline pkg/hintrunner/zero/hintcode.go:41]: Missing the Code suffix in the const name.

File: pkg/utils/arrays.go [hunk 2]
```diff
@@ -5,3 +5,10 @@ func Reverse[T any](a []T) {
 		a[i], a[j] = a[j], a[i]
 	}
 }
+
```
Comment (quasilyte) [inline pkg/utils/arrays.go:8]: I would suggest using the stdlib function for that.
https://pkg.go.dev/slices#Reverse

Comment (MaksymMalicki) [reply pkg/utils/arrays.go]: Didn't notice the stdlib function. Replaced

File: pkg/hintrunner/zero/hintcode.go (hunk 1 above)
Comment (MaksymMalicki) [reply pkg/hintrunner/zero/hintcode.go]: Added code suffix

File: pkg/hintrunner/zero/zerohint_uint256_test.go [hunk 3]
```diff
@@ -131,5 +131,67 @@ func TestZeroHintUint256(t *testing.T) {
...
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newUint256SqrtHint(ctx.operanders["n.low"], ctx.operanders["root.low"])
+				},
+				check: allVarValueEquals(map[string]*fp.Element{
+					"root.low":  feltUint64(4),
+					"root.high": feltUint64(0),
+				}),
+			},
+		},
 	})
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_uint256_test.go:196]: I think this last test case is not necessary as you are already checking with another value that is simple (the previous case). I'd rather you add a test for a bigger number, one that doesn't fit in a felt either, just in a uint256. Other test case that could be useful would be one in which `n` is not a square.

Comment (MaksymMalicki) [reply pkg/hintrunner/zero/zerohint_uint256_test.go]: Added test cases for bigger numbers, this was a very good call, as I have found critical error, so TY

File: pkg/hintrunner/zero/zerohint_uint256.go [hunk 4]
```diff
@@ -190,3 +192,64 @@ func createSplit64Hinter(resolver hintReferenceResolver) (hinter.Hinter, error)
...
+				return err
+			}
+			//> ids.root.low = root
+			rootLowValue := memory.MemoryValueFromFieldElement(&calculatedFeltRoot)
+			err = vm.Memory.WriteToAddress(&rootAddr, &rootLowValue)
+			if err != nil {
+				return err
+			}
+			//> ids.root.high = 0
+			rootHighValue := memory.MemoryValueFromFieldElement(new(fp.Element).SetInt64(0))
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_uint256.go:239]: Could you use FeltZero constant from utils here?

File: pkg/hintrunner/zero/zerohint_uint256.go [hunk 5]
```diff
@@ -190,3 +192,64 @@ func createSplit64Hinter(resolver hintReferenceResolver) (hinter.Hinter, error)
...
+			return hinter.WriteToNthStructField(vm, root, rootHighValue, 1)
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_uint256.go:240]: If we merge #308 first then you can use `WriteUint256ToAddress` utility here for writing `root` to memory

Comment (MaksymMalicki) [reply pkg/hintrunner/zero/zerohint_uint256.go:240]: yes, absolutely

File: pkg/hintrunner/zero/zerohint_uint256.go (hunk 4 above)
Comment (MaksymMalicki) [reply pkg/hintrunner/zero/zerohint_uint256.go:239]: done

File: pkg/hintrunner/zero/zerohint_uint256.go [hunk 6]
```diff
@@ -190,3 +192,64 @@ func createSplit64Hinter(resolver hintReferenceResolver) (hinter.Hinter, error)
 
 	return newSplit64Hint(a, low, high), nil
 }
+
+func newUint256SqrtHint(n hinter.ResOperander, root hinter.ResOperander) hinter.Hinter {
+	// from starkware.python.math_utils import isqrt\nn = (ids.n.high << 128) + ids.n.low\nroot = isqrt(n)\nassert 0 <= root < 2 ** 128\nids.root.low = root\nids.root.high = 0
```
[the remark this answers is not in the document]
Comment (MaksymMalicki) [reply pkg/hintrunner/zero/zerohint_uint256.go:197]: done

**Review by cicr99 — APPROVED** [review]

**Pull request merged**